open-metadata-implementation/access-services/subject-area/subject-area-spring/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/server/spring/SubjectAreaCategoryRESTResource.java

```diff
@@ -112,17 +115,18 @@ public  SubjectAreaOMASAPIResponse getCategoryByGuid(@PathVariable String server
      * <li> FunctionNotSupportedException        Function not supported this indicates that a find was issued but the repository does not implement find functionality in some way.</li>
      * </ul>
      */
-    @GetMapping( path = "/users/{userId}/categories")
-    public  SubjectAreaOMASAPIResponse findTerm(@PathVariable String serverName, @PathVariable String userId,
-                                                @RequestParam(value = "searchCriteria", required=false) String searchCriteria,
-                                                @RequestParam(value = "asOfTime", required=false) Date asOfTime,
-                                                @RequestParam(value = "offset", required=false) Integer offset,
-                                                @RequestParam(value = "pageSize", required=false) Integer pageSize,
-                                                @RequestParam(value = "sequencingOrder", required=false) SequencingOrder sequencingOrder,
-                                                @RequestParam(value = "sequencingProperty", required=false) String sequencingProperty
-    )  {
-        return restAPI.findCategory(serverName,userId,searchCriteria,asOfTime,offset,pageSize,sequencingOrder,sequencingProperty);
+    @GetMapping(path = "/users/{userId}/categories")
+    public SubjectAreaOMASAPIResponse findTerm(@PathVariable String serverName, @PathVariable String userId,
+                                               @RequestParam(value = "searchCriteria", required = false) String searchCriteria,
+                                               @RequestParam(value = "asOfTime", required = false) Date asOfTime,
+                                               @RequestParam(value = "offset", required = false, defaultValue = PAGE_OFFSET_DEFAULT_VALUE) Integer offset,
+                                               @RequestParam(value = "pageSize", required = false, defaultValue = PAGE_SIZE_DEFAULT_VALUE) Integer pageSize,
```

Comment:
By setting a default page size- this means that paging cannot be disabled in the API, which means this is an API change. This means the caller would be expected to deal with paging but has not requested paging. I am therefor thinking that we should not default this value. The same for the other find requests. I think it is ok to default offset to 0.

Reply:
Let's set PAGE_SIZE_DEFAULT_VALUE = 0, as it is done in the current version only elsewhere) and have "Zero means unrestricted return results size." as indicated in the specification.

Reply:
@immmus I see PAGE_SIZE_DEFAULT_VALUE = "100"; currently in this change . I am going to do further test / reviewing on Monday. Thanks for all your efforts with this.

Reply:
Yes, is my currently in this change PAGE_SIZE_DEFAULT_VALUE = "100"; 
But I suggest changing to 0 because it fits the specification. Should I commit with this change ??

Reply:
I decided do not wait your answer and commit this change:). Since judging by our conversation, I think we agreed on a common opinion.